ios/RCTWebRTC/WebRTCModule+Daily.m

```diff
@@ -139,22 +124,30 @@ - (void)audioSession:(RTCAudioSession *)audioSession didSetActive:(BOOL)active {
     return;
   }
   
-  self.audioMode = audioMode;
+  [self setAudioMode: audioMode];
   
   // Apply the chosen audio mode right away if the audio session is already
   // active. Otherwise, it will be applied when the session becomes active.
   RTCAudioSession *audioSession = RTCAudioSession.sharedInstance;
+  NSLog(@"[Daily] setDailyAudioMode: %@", audioMode);
   if (audioSession.isActive) {
     [self applyAudioMode:audioMode toSession:audioSession];
   }
 }
 
 - (void)applyAudioMode:(NSString *)audioMode toSession:(RTCAudioSession *)audioSession {
+  NSLog(@"[Daily] applyAudioMode: %@", audioMode);
   // Do nothing if we're attempting to "unset" the in-call audio mode (for now
   // it doesn't seem like there's anything to do).
   if ([audioMode isEqualToString:AUDIO_MODE_IDLE]) {
     return;
   }
+    
+  if ([audioMode isEqualToString:AUDIO_MODE_USER_SPECIFIED_ROUTE]) {
+    // Invoking to restore to the user chosen device
+    [self setAudioDevice:self.userSelectedDevice];
+    return;
+  }
   
```

Comment:
In the case of the customer going from `AUDIO_MODE_USER_SPECIFIED_ROUTE` -> either `AUDIO_MODE_VIDEO_CALL` or `AUDIO_MODE_VOICE_CALL`, we might need to "undo" the changes that `setAudioDevice:` does, like force speaker and set preferred device...

...or maybe we don't need to!

Reply:
I guess the best way to test this is with the scenario where the user:
* Explicitly chooses speakerphone
* Then switches back to "voice" audio mode

Reply:
It works as expected, when we switch to voice mode, it switches the audio route to earpiece.